src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useCallback, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { ErrorNoticeType } from '../../types/ErrorNoticeType';
+
+interface Props {
+  newTodoField: React.RefObject<HTMLInputElement>,
+  isAdding: boolean,
+  isToggleVisible: boolean,
+  isAllTodosCompleted: boolean,
+  setHasError: (status: boolean) => void,
+  setErrorNotice: (notice: ErrorNoticeType) => void,
+  postTodoToServer: (title: string) => void,
+  toggleAllTodosStatus: () => void,
+}
+
+export const NewTodo: React.FC<Props> = ({
+  newTodoField,
+  isAdding,
+  isToggleVisible,
+  isAllTodosCompleted,
+  setHasError,
+  setErrorNotice,
+  postTodoToServer,
+  toggleAllTodosStatus,
+}) => {
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!newTitle.trim()) {
+        setHasError(true);
+        setErrorNotice(ErrorNoticeType.TitleError);
+
+        return;
+      }
+
+      postTodoToServer(newTitle);
+    }, [newTitle],
+  );
+
+  useEffect(() => {
+    if (!isAdding) {
+      setNewTitle('');
+    }
```

Comment:
What if something goes wrong? Don't clear input before you get an answer from server